Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import CoreCommands
+import TSCUtility
+import TSCBasic
+import PackageModel
+import Foundation
+
+extension SwiftPackageTool {
+    
+    struct Install: SwiftCommand {
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(abstract: "Offers the ability to install executable targets of the current package.")
+        }
+        
+        @OptionGroup()
+        var globalOptions: GlobalOptions
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.getOrCreateDotSwiftPMDirectory()
+            
+            let env = ProcessInfo.processInfo.environment
+            
+
+            // ~/.swiftpm/bin/
+            let swiftpmBinDir = dotSwiftPMDir.appending(component: "bin")
```

Comment:
I don't see where you check whether this directory exists and create it if it doesn't. Would you mind moving this path calculation and creation logic to `FileSystem+Extensions.swift` so that it matches what we already do for other paths in `~/.swiftpm`? Something like `FileSystem.getOrCreateSwiftPMInstalledBinariesDirectory()`?